sourcegraph/learn
# Add styled-components and remove bootstrap

### What should this PR do?
Resolves [DEVED-19](https://sourcegraph.atlassian.net/browse/DEVED-19) by implementing `styled-components` and removing Bootstrap.

Also resolves the remaining linting issues noted in [DEVED-20](https://sourcegraph.atlassian.net/browse/DEVED-20)

### Why are we making this change?
We have decided to overhaul our CSS strategy — see the [decision doc](https://docs.google.com/document/d/1_SR-p_nXXLUD7Nv8SlW9a5jjCHWAFuw2-nupCDUqDp8) for more details. TL;DR — we have decided to adopt `styled-components` to implement a more modular, scoped approach to styles across the app, in line with the thinking outlined [here](https://docs.google.com/document/d/1VedPoFy7WIuHeXPzwRGMmgjDPL4CBdTKmhy4OBmlmf8), and looking towards the layout component work detailed [here](https://docs.google.com/document/d/1wdQXdxz4BukG0vQLF_3v9gXYLiksYjLUQlyRUwtMVKg). Though this PR does not adopt the exact same components outlined in this last RFC, it is thinking in those terms.

### What are the acceptance criteria? 
- Styles on the site look as they do on production, with the exception of certain changes that were made in the spirit of optimization.

### How should this PR be tested?
- Check out the branch, and do a fresh install of dependencies, by nuking `node_modules` and running `npm ci` to install the pinned dependencies outlined in `package-lock.json`.
- Check the following on both desktop and mobile: home page, post pages, tag pages, nav behavior, footer behavior, author index.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
